pandas/core/internals/blocks.py

```diff
@@ -601,7 +601,7 @@ def replace(
             return blocks
 
         elif self.ndim == 1 or self.shape[0] == 1:
-            blk = self.coerce_to_target_dtype(value)
+            blk = self.coerce_to_target_dtype(value, strict_na=True)
```

Comment:
ATM i think rather than add special cases to the lower-level functions, we should keep it isolated here, similar to what you did for #45836 (and ideally down the line combine both fixes into something more elegant)

Reply:
sure.

from https://github.com/pandas-dev/pandas/issues/45725#issuecomment-1073392541

> The alternative is to intercept this particular situation _somewhere_. I think @simonjayhawkins has a branch that intercepts it at the level of find_common_type, 

I think the issues arises basically from the difference between _can_hold_element and is_valid_na_for_dtype not being strict (i.e. None and pd.NA are considered valid for float array).

I have added a strict flag to pass through to the lower level library casting functions (i've put this in `find_result_type` for now since that's where the code was changed arising in the regression reported here.

But it probably does make more sense that the flag is passed further down through to is_valid_na_for_dtype and then maybe use _can_hold_element in is_valid_na_for_dtype when the strict flag is True.

also from https://github.com/pandas-dev/pandas/issues/45725#issuecomment-1073392541

> but unless there are other use cases I'm inclined to handle it in Block.replace

Seems reasonable. I'm not sure if there are any other use cases. I suspect we need some logic that uses is_valid_na_for_dtype in Block.replace for this to work but also suspect that there will also be some duplication with the lower level library functions to pull this off.

As mentioned above, would also be quite happy doing this as a refactor after the recursion issue is resolved by the fix suggested here.

> rather than add special cases to the lower-level functions

I've not yet looked but off the top of your head, is the code added in https://github.com/pandas-dev/pandas/pull/45568 to the lower level functions a special case for Interval? The reason I ask is removing that would also fix the recursion error.

Reply:
> I've not yet looked but off the top of your head, is the code added in https://github.com/pandas-dev/pandas/pull/45568 to the lower level functions a special case for Interval? The reason I ask is removing that would also fix the recursion error.

Most of the edits in #45568 should be correct for The General Case.  The only special case for IntervalDtype is in ensure_dtype_can_hold_na, which I don't think is what you're referring to.